doc/source/developer.rst

```diff
@@ -50,15 +50,36 @@ So that a ``pandas.DataFrame`` can be faithfully reconstructed, we store a
     'pandas_version': $VERSION}
 
 Here, ``<c0>``/``<ci0>`` and so forth are dictionaries containing the metadata
-for each column. This has JSON form:
+for each column, *including the index columns*. This has JSON form:
 
 .. code-block:: text
 
    {'name': column_name,
+    'field_name': arrow_table_column_name,
     'pandas_type': pandas_type,
     'numpy_type': numpy_type,
     'metadata': metadata}
 
+.. note::
+
+   The last ``N`` values of ``metadata['columns']``, where ``N =
+   len(metadata['index_columns'])``, contain information about the row indexes,
+   including the name of the index level and type information.
```

Comment:
Do we want this in the spec? (meaning, do we expect fastparquet to change to follow this). It's nice to have it consistent accross libraries, but not that important for a faithful roundtrip anymore with the introduction of "field_name"

Reply:
I think removing this is a good idea.